cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java

```diff
@@ -704,6 +707,60 @@ public Void apply(AppMds mds) throws Exception {
     });
   }
 
+  @Override
+  public void deleteSchedules(final Id.Program program) {
+    txnl.executeUnchecked(new TransactionExecutor.Function<AppMds, Void>() {
+      @Override
+      public Void apply(AppMds mds) throws Exception {
+        ApplicationSpecification existingAppSpec = getAppSpecOrFail(mds, program);
+        ApplicationSpecification newAppSpec =
+          removeSchedulesFromApp(existingAppSpec, new Predicate<ScheduleProgramInfo>() {
+            @Override
+            public boolean apply(@Nullable ScheduleProgramInfo scheduleProgramInfo) {
+              return scheduleProgramInfo != null && scheduleProgramInfo.getProgramName().equals(program.getId()) &&
+                scheduleProgramInfo.getProgramType().equals(program.getType().getSchedulableType());
+            }
+          });
+        replaceAppSpecInProgramJar(program, newAppSpec);
+        mds.apps.updateAppSpec(program.getNamespaceId(), program.getApplicationId(), newAppSpec);
+        return null;
+      }
+    });
+  }
+
+  @Override
+  public void deleteSchedules(final Id.Namespace namespaceId) {
+    txnl.executeUnchecked(new TransactionExecutor.Function<AppMds, Void>() {
+      @Override
+      public Void apply(AppMds mds) throws Exception {
+        for (ApplicationSpecification appSpec : getAllApplications(namespaceId)) {
+          ApplicationSpecification newAppSpec =
+            removeSchedulesFromApp(appSpec, new Predicate<ScheduleProgramInfo>() {
+              @Override
+              public boolean apply(@Nullable ScheduleProgramInfo scheduleProgramInfo) {
+                return true;
+              }
+            });
+//          replaceAppSpecInProgramJar(program, newAppSpec);
+          mds.apps.updateAppSpec(namespaceId.getId(), appSpec.getName(), newAppSpec);
+        }
+        return null;
+      }
+    });
+  }
+
+  private ApplicationSpecification removeSchedulesFromApp(ApplicationSpecification appSpec,
+                                                          final Predicate<ScheduleProgramInfo> predicate) {
```

Comment:
Just to make things clear, can you call `predicate` as `removalPredicate` or something similar?